### PR TITLE
Better init

### PR DESCRIPTION
**Pull request by josegonzalez**
From comments in #3, I decided to take a stab at updating paths to be more "correct".

Seems like tests fail, potentially due to permissions, but I'm not sure at this time...

### CODE REVIEW

**Comment by joshmcrae** [comment]
I've removed variable expansion from the env stanzas. The environment variables can still be overridden when Upstart is used to start the service manually.

The socket was also being placed in `/var/sock/dokku-daemon/dokku-daemon.sock`.

**Comment by josegonzalez** [comment]
Heh oops. Do you want me to squash everything? If not, anything else this needs before merge?

File: init/dokku-daemon.conf [hunk 1]
```diff
@@ -6,9 +6,33 @@ author "Josh McRae"
...
+  if [ "${DOKKU_LOCK_DIR}" != "/tmp" ]; then
+    mkdir -p "${DOKKU_LOCK_DIR}" || true
+    chmod $PERMS "${DOKKU_LOCK_DIR}" || true
+    chown $USER:$GROUP "${DOKKU_LOCK_DIR}" || true
+  fi
+  if [ "${DOKKU_SOCK_DIR}" != "/tmp" ]; then
+    mkdir -p "${DOKKU_SOCK_DIR}" || true
+    chmod $PERMS "${DOKKU_SOCK_DIR}" || true
+    chown $USER:$GROUP "${DOKKU_SOCK_DIR}" || true
+  fi
```
Comment (josegonzalez) [inline init/dokku-daemon.conf:28]: Don't know if the above can be done any nicer...

**Comment by joshmcrae** [comment]
This looks way better - there's nothing I can see that needs to be added right now. I'll have a look at creating a systemd init soon.

Could you please squash these commits

File: init/dokku-daemon.conf (hunk 1 above)
Comment (joshmcrae) [inline init/dokku-daemon.conf:28]: `mkdir` has the -m flag which could replace usage of `chmod`?

Comment (josegonzalez) [inline init/dokku-daemon.conf:28]: Good call, I'll do that when I get home.

**Comment by joshmcrae** [comment]
@josegonzalez Do you need anything further from me? I'm happy for this to be merged.

**Comment by josegonzalez** [comment]
Merged! FWIW, I'm `savant` on our irc channel and on [slack](https://glider-slackin.herokuapp.com/).

**Pull request merged**